### ISSUE TITLE
invalid argument "-std=c99" not allowed with "C++"

### ISSUE THREAD

**Issue by kapitan-iglu** [CoatiSoftware/Sourcetrail#1016]
Hi,
I have similar problem to #937 but slightly different (therefore new issue). 

I'm trying to create new project with only C source for Texas Instruments [TMS320C24xx](https://www.ti.com/product/TMS320LF2407A) processor. 
- I've created `New Source Group` from `Empty C Source Group`
- Then I configured paths and set `C Standard` to `C99` (but same result also for C11, C90, C89)
- `Cross-Compilation` unchecked.

After indexing for all `*.c` files same error is generated:

    ERROR invalid argument ''-std=c99'' not allowed with ''C++''

Why is clang trying to compile my C source as C++?
What I'm doing wrong?

Sourcetrail Version 2020.01.117 - 64bit, DB Ver 25, Windows 7

**Comment by mlangkabel** [comment]
This is strange. I just tried, but cannot reproduce the issue. Did you add anything in the "Compiler Flags" list of your Sourcetrail project? For example if I create a C source group but add "-xc++" as a compiler flag, this will override the language standard again.

**Comment by kapitan-iglu** [comment]
> Did you add anything in the "Compiler Flags" list of your Sourcetrail project?

Yes, but nothing like that. I pass only my custom macro `-DSOURCETRAIL` to the compiler to hide embedded compiler specific keyword `ioport` which is unknown for clang.

I stripped source files from project step by step and I ended with only one empty(!) C file generating `std=c99 not allowed with C++` error.

So, I created empty folder with only one empty file named `MAIN.C`. Then I created new Sourcetrail project from `Empty C Source Group` with following settings:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<config>
    <source_groups>
        <source_group_8e30cdfb-1c76-4984-ac20-0405c1862329>
            <c_standard>c99</c_standard>
            <cross_compilation>
                <target>
                    <abi>unknown</abi>
                    <arch>x86_64</arch>
                    <sys>unknown</sys>
                    <vendor>unknown</vendor>
                </target>
                <target_options_enabled>0</target_options_enabled>
            </cross_compilation>
            <name>C Source Group</name>
            <pch_flags>
                <use_compiler_flags>1</use_compiler_flags>
            </pch_flags>
            <source_extensions>
                <source_extension>.c</source_extension>
            </source_extensions>
            <source_paths>
                <source_path>..</source_path>
            </source_paths>
            <status>enabled</status>
            <type>C Source Group</type>
        </source_group_8e30cdfb-1c76-4984-ac20-0405c1862329>
    </source_groups>
    <version>8</version>
</config>
```

After indexing `std=c99 not allowed with C++` error appears. Then I renamed `MAIN.C` to `MAIN.c` and after re-indexing error disappears. So I guess that clang is considering files with uppercase extension `C` as C++...

Is this behavior bug or feature? (from my point of view is bug)

**Comment by kapitan-iglu** [comment]
As documented [here](http://gcc.gnu.org/onlinedocs/gcc-4.4.1/gcc/Overall-Options.html#index-file-name-suffix-71) gcc (and obviously also clang) is detecting language based on filename extension (and yes, `C` means C++). So I solved problem by adding language switch `-xc` to compiler flags.

Is it possible to force clang language type based on source group type in Sourcetrail?

So, if I create C source group then `-x c` will be prepended to additional compiler flags. For C++ group `-x c++` and so on... 

When I select exact type of source group in Sourcetrail I expect that every file within this group will be considered as that type regardless of file extension.

**Comment by mlangkabel** [comment]
Ha! We are already adding `-x c++` for c++ sourcegroups! But we are not doing anything for C.

**Comment by mlangkabel** [comment]
Fixed by merge of pull request #1038
